dev-tools/reconfigurator-exec/src/main.rs

```diff
@@ -0,0 +1,175 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Execute blueprints from the command line
+
+use anyhow::Context;
+use anyhow::anyhow;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db;
+use nexus_db_queries::db::DataStore;
+use nexus_reconfigurator_execution::realize_blueprint;
+use nexus_types::deployment::UnstableReconfiguratorState;
+use omicron_uuid_kinds::BlueprintUuid;
+use omicron_uuid_kinds::OmicronZoneUuid;
+use slog::info;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use update_engine::EventBuffer;
+use update_engine::NestedError;
+use update_engine::display::LineDisplay;
+use update_engine::display::LineDisplayStyles;
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let args = ReconfiguratorExec::parse();
+
+    if let Err(error) = args.exec().await {
+        eprintln!("error: {:#}", error);
+        std::process::exit(1);
+    }
+
+    Ok(())
+}
+
+/// Execute blueprints from the command line
+#[derive(Debug, Parser)]
+struct ReconfiguratorExec {
+    /// log level filter
+    #[arg(
+        env,
+        long,
+        value_parser = parse_dropshot_log_level,
+        default_value = "info",
+        global = true,
+    )]
+    log_level: dropshot::ConfigLoggingLevel,
+
+    dns_server: SocketAddr,
+
+    // XXX-dap assumed unused
+    #[arg(long, default_value = "080c53b8-7833-45c6-bfc3-63466a416228")]
+    nexus_id: OmicronZoneUuid,
```

Comment:
I want to remove this.  Right now, if you had an expunged Nexus zone, I think this command would re-assign that Nexus's sagas to this id, which is basically abandoning them (and totally wrong).  I'm inclined to make this an optional argument to `realize_blueprint` instead and just skip saga re-assignment.

(Alternatively, eventually, we're going to have to make saga re-assignment aware of zone images, and then it will no longer be correct for Nexus to assign sagas to itself.  Instead, it will have to pick a running Nexus with the same image.  Once it does that, this argument would go away entirely.)

Reply:
Done.